advanced/Scripts/list.sh

```diff
@@ -21,68 +21,71 @@ web=false
 
 domList=()
 
-listType=""
-listname=""
+typeId=""
 
 colfile="/opt/pihole/COL_TABLE"
 source ${colfile}
 
+GetListnameFromTypeId() {
+    if [[ "$1" == "0" ]]; then
+        echo "whitelist"
+    elif  [[ "$1" == "1" ]]; then
+        echo "blacklist"
+    elif  [[ "$1" == "2" ]]; then
+        echo "regex_whitelist"
+    elif  [[ "$1" == "3" ]]; then
+        echo "regex_blacklist"
+    fi
+}
 
-helpFunc() {
-    if [[ "${listType}" == "whitelist" ]]; then
-        param="w"
-        type="whitelist"
-    elif [[ "${listType}" == "regex_blacklist" && "${wildcard}" == true ]]; then
-        param="-wild"
-        type="wildcard blacklist"
-    elif [[ "${listType}" == "regex_blacklist" ]]; then
-        param="-regex"
-        type="regex blacklist filter"
-    elif [[ "${listType}" == "regex_whitelist" && "${wildcard}" == true ]]; then
-        param="-white-wild"
-        type="wildcard whitelist"
-    elif [[ "${listType}" == "regex_whitelist" ]]; then
-        param="-white-regex"
-        type="regex whitelist filter"
-    else
-        param="b"
-        type="blacklist"
+GetListParamFromTypeId() {
+    if [[ "${typeId}" == "0" ]]; then
+        echo "w"
+    elif  [[ "${typeId}" == "1" ]]; then
+        echo "b"
+    elif  [[ "${typeId}" == "2" && "${wildcard}" == true ]]; then
+        echo "-white-wild"
+    elif  [[ "${typeId}" == "2" ]]; then
+        echo "regex_blacklist"
+    elif  [[ "${typeId}" == "3" && "${wildcard}" == true ]]; then
+        echo "-regex"
+    elif  [[ "${typeId}" == "3" ]]; then
+        echo "-wild"
     fi
+}
+
+helpFunc() {
+    local listname param
+
+    listname="$(GetListnameFromTypeId "${typeId}")"
+    param="$(GetListParamFromTypeId)"
 
     echo "Usage: pihole -${param} [options] <domain> <domain2 ...>
 Example: 'pihole -${param} site.com', or 'pihole -${param} site1.com site2.com'
-${type^} one or more domains
+${listname^} one or more domains
```

Comment:
`pihole --regex -h` will show (actually untested, just assuming from the code):
```
Regex_blacklist one or more domains
```

We should consider replacing the underscore by a space on the fly.

Reply:
`${listname/_/ /}` should do it if `listname` doesn't need to be converted to capitals with `^`.

Reply:
Actually, no need to replace the space on the fly. It is only ever used in output, so might as well just have a space in the string to start with.